scrapy/core/spidermw.py

```diff
@@ -50,21 +55,36 @@ def process_spider_input(response):
 
         def process_spider_exception(_failure):
             exception = _failure.value
+            # ignore AssertionError from middleware's manager
+            if isinstance(exception, AssertionError) and len(exception.args) > 0 \
```

Comment:
You're right, that would be way less hacky than checking the exception message. I'm on it :-)